lib/fedora_migrate/datastream_verification.rb

```diff
@@ -4,28 +4,20 @@ module FedoraMigrate::DatastreamVerification
 
   def valid? datastream=nil
     @datastream = datastream || @source
-    check = has_no_checksum? || has_matching_checksums? || content_equivalent?
+    check = has_matching_checksums? || has_matching_nokogiri_checksums?
     FedoraMigrate::Logger.warn "#{@datastream.pid} datastream #{@datastream.dsid} validation failed" unless check
     check
   end
 
   def has_matching_checksums?
-    datastream.checksum == target_checksum
+    datastream.checksum == target_checksum || Digest::SHA1.hexdigest(datastream.content) == target_checksum
   end
 
-  # TODO: Reporting mechanism? If there isn't a checksum it defaults to "none" (issue #4)
-  def has_no_checksum?
-    if datastream.checksum == "none"
-      FedoraMigrate::Logger.info "unable to vaidate datastream because the checksum was 'none'"
-      return true
-    end
-  end
-  
-  # TODO: In some cases, the line <?xml version="1.0"?> is being added to migrated xml datastreams.
-  # This invalidates the checksum even though the xml content is the same. (issue #11)
-  def content_equivalent?
+  def has_matching_nokogiri_checksums?
     if datastream.mimeType == "text/xml"
-      EquivalentXml.equivalent?(target_content, datastream.content)
+      Digest::SHA1.hexdigest(Nokogiri::XML(datastream.content).to_xml) == Digest::SHA1.hexdigest(Nokogiri::XML(target_content).to_xml)
```

Comment:
How about creating a method named something like `checksum` in this file or elsewhere if there's a better place in the fedora-migrate codebase? That way you don't have to type "Digest::SHA1.hexdigest" over and over again, and it also makes it easier to folks to override this value with their own checksumming method.